src/components/Event/EventBox.tsx

```diff
@@ -0,0 +1,115 @@
+export interface EventBoxProps {
+  EventName: string;
+  Enroll: boolean;
+  CurrentEnrollment: number;
+  MaxEnrollment: number;
+  ThaiFaculty: string;
+  EngFaculty: string;
+  Date: string;
+  Time: string;
+  Location: string;
+}
+const EnrollBox: React.FC<{
+  CurrentEnrollment: number;
+  MaxEnrollment: number;
+}> = ({ CurrentEnrollment, MaxEnrollment }) => (
+  <div className="flex flex-row gap-2 text-white">
+    <div className="rounded-full bg-pink-500 px-2 py-1 ">
+      <p className="text-xs">ต้องลงทะเบียน/ Limited seats</p>
+    </div>
+    <div className="rounded-full border-2 border-white px-2 py-1">
+      <p className="text-xs">
+        {CurrentEnrollment} / {MaxEnrollment}
+      </p>
+    </div>
+  </div>
+);
+
+const EventBox = (props: EventBoxProps) => {
+  return (
+    <div className="box-border w-72 rounded-lg border-2 border-white bg-gradient-to-br from-gray-100 to-pink-400 shadow-md ring-offset-2 ring-offset-white">
+      <a href="\">
+        <div className="border-1 rounded-md border-black bg-gradient-to-r from-[#393570]  to-[#CA4072] px-4 py-2">
+          <p className="font-ibm text-xl font-semibold uppercase text-white">
+            {props.EventName}
+          </p>
+          {props.Enroll && (
+            <EnrollBox
+              CurrentEnrollment={props.CurrentEnrollment}
+              MaxEnrollment={props.MaxEnrollment}
+            />
```

Comment:
There is another way to pass a prop from a variable:

```suggestion
            <EnrollBox
              {...{MaxEnrollment, CurrentEnrollment}}
            />
```
In my opinion, the above code is more comprehensive about which variables are passed as props to the EnrollBox component.

Since this is only my opinion, feel free to discard my suggestion if it doesn't sound good to you. and feel free to state any arguments or questions here.